include/deal.II/lac/trilinos_tpetra_precondition.h

```diff
@@ -1316,9 +1316,192 @@ namespace LinearAlgebra
+    class PreconditionAMGMueLu : public PreconditionBase<Number, MemorySpace>
+    {
+    public:
+      /**
+       * The chosen smoother is not supported or configured with Ifpack2.
+       */
+      DeclException1(
+        ExcTrilinosMueLuSmootherUnsupported,
+        std::string,
+        << "You tried to select the smoother type <" << arg1 << ">\n"
+        << "but this is not supported by our interface\n"
+        << "due to one of the following reasons:\n"
+        << "* This smoother does not exist\n"
+        << "* This smoother is not (yet) supported by our interface\n"
+        << "* This smoother is not (yet) supported by MueLu.\n");
+      DeclException1(
+        ExcTrilinosMueLuCoarseSolverUnsupported,
+        std::string,
+        << "You tried to select the coarse solver type <" << arg1 << ">\n"
+        << "but this is not supported by our interface\n"
+        << "due to one of the following reasons:\n"
+        << "* This coarse solver does not exist\n"
+        << "* This coarse solver is not (yet) supported by our interface\n"
+        << "* This coarse solver is not (yet) supported by MueLu.\n");
+
+      /**
+       * @brief The set of additional parameters to tune the preconditioner.
+       *
+       */
+      struct AdditionalData
+      {
+        /**
+         * @brief Constructor.
+         *
+         * By default, we pretend to work on elliptic problems with linear
+         * finite elements on a scalar equation.
+         *
+         * @param elliptic Optimize MueLu for elliptic problems?
+         * @param symmetric Assume for A to be symmetric?
+         * @param w_cycle Use W-cycle instead of V-cycle?
+         * @param aggregation_threshold Threshold for coarsening.
+         * @param constant_modes Specify constant modes (near null space) of the matrix.
+         * @param smoother_sweeps Number of times to apply the smoother.
+         * @param smoother_overlap Overlap of smoother if run in parallel.
+         * @param output_details Print additional info to screen?
+         * @param smoother_type Determine the smoother to use.
+         * @param coarse_type Determine the coarse solver.
+         * @param relaxation_type Determine the specific smoother for type RELAXATION.
+         */
+        AdditionalData(const bool         elliptic              = true,
+                       const bool         symmetric             = true,
+                       const bool         w_cycle               = false,
+                       const double       aggregation_threshold = 1e-4,
+                       const int          smoother_sweeps       = 2,
+                       const int          smoother_overlap      = 0,
+                       const bool         output_details        = false,
+                       const std::string &smoother_type         = "Chebyshev",
+                       const std::string &coarse_type           = "KLU2");
```

Comment:
`KLU2` will require Amesos2 to be activated. Is this guaranteed when running MueLu & Tpetra?

Reply:
For the old wrappers we required Teuchos, Epetra, Amesos, Ifpack, AztecOO and ML
and we discussed to add the matching requirement for Tpetra, i.e. Teuchos, Amesos2, Ifpack2, Belos and MueLu.
But for now I could add an additional guard for Amesos2 to the MueLu file.

Reply:
Requiring the mentioned list of packages seems reasonable. I was just wondering, if this implicit assumption of Amesos2 being activated is somehow made explicit at some point.